src/Sources/AbstractSource.php

```diff
@@ -135,24 +135,25 @@ function ($migration_class) {
         $this->onUnInstall();
     }
 
-    protected function save(Model $me, Model $primary)
+    protected function save(Model $source, $primary = null)
     {
-        // @todo: update $me name
         // @todo: Only save model if something has changed
         // @todo: in the Audit, describe what changed
+
         // Now, we save the primary version
-        $primary->save();
+        if($primary !== null){
+            $primary->save();
+            $source->game_id = $primary->id;
```

Comment:
Yeah, this is something I did that was wrong. We'll have to figure something out here.

`game_id` works for us, but that won't be the column name for others. And the framework can't care about those things.

So, I think we have 2 options:
1. Require the source itself to populate this column
2. Require the source to declare what the column name is. Add another `abstract function getPrimaryEntityJoinColumnName()` or something like that. Then, here we can 

```php
$column_name = static::getPrimaryEntityJoinColumnName();
$source->{$column_name} = $primary->id;
```

Reply:
So why can't the other sources also have a ``game_id`` column? Isn't that what a one-to-many looks like?

Reply:
It's not that the other **sources** can't. In fact, they will.

It's that other **projects** won't. If the framework is used to collect financial data for car sales, the primary entity will be something like `cars` and each source will want to have a `car_id`, not a `game_id`

Everything in `AbstractSource` needs to be blind to what kind of project this is.